src/memory/memory.cc

```diff
@@ -1,44 +1,74 @@
-#include <algorithm>
-
 #include "memory/memory.hh"
 
 namespace sim {
 
 Word Memory::loadWord(Addr addr) {
   stats.numLoads++;
-  Word word = kDummyWord;
-
-  auto it = mem.find(addr);
-  if (it != mem.end())
-    word = it->second;
-  else
-    word = pageFaultHandle(addr);
+  Word loadedWord = physMem->getWord(addr, PhysMemory::MemoryOp::LOAD);
 
-  return word;
+  return loadedWord;
 }
 
 void Memory::storeWord(Addr addr, Word word) {
   stats.numStores++;
-  mem[addr] = word;
-}
-
-Word Memory::pageFaultHandle(Addr addr) {
-  (void)addr;
-  stats.numPageFaults++;
-  std::cout << "Placeholder for page fault" << std::endl;
-  return kDummyWord;
+  physMem->getWord(addr, PhysMemory::MemoryOp::STORE) = word;
 }
 
-std::size_t Memory::getCurrMemSize() const { return mem.size(); }
-
 void Memory::printMemStats(std::ostream &ost) const {
   ost << "Memory stats:" << std::endl;
   ost << "Loads: " << stats.numLoads << std::endl;
   ost << "Stores: " << stats.numStores << std::endl;
-  ost << "PageFaults: " << stats.numPageFaults << std::endl;
   ost << "End of memory stats." << std::endl;
 }
 
 const Memory::MemoryStats &Memory::getMemStats() const { return stats; }
 
+PhysMemory::AddrSections PhysMemory::getVirtAddrSections(Addr addr) {
```

Comment:
I suppose it's better to transform this function into constructor of `AddrSections` class

Reply:
Done